ts/client/src/accounts/oracle.ts

```diff
@@ -48,21 +48,29 @@ export class StubOracle {
 }
 
 // https://gist.github.com/microwavedcola1/b741a11e6ee273a859f3ef00b35ac1f0
-export function parseSwitcboardOracleV1(
-  accountInfo: AccountInfo<Buffer>,
-): number {
-  return accountInfo.data.readDoubleLE(1 + 32 + 4 + 4);
+export function parseSwitcboardOracleV1(accountInfo: AccountInfo<Buffer>): {
+  price: number;
+  lastUpdatedSlot: number;
+} {
+  const price = accountInfo.data.readDoubleLE(1 + 32 + 4 + 4);
+  const lastUpdatedSlot = parseInt(
+    accountInfo.data.readBigUInt64LE(1 + 32 + 4 + 4 + 8).toString(),
+  );
+  return { price, lastUpdatedSlot };
 }
 
 export function parseSwitcboardOracleV2(
   program: SwitchboardProgram,
   accountInfo: AccountInfo<Buffer>,
-): number {
-  const aggregatorAccountData =
-    program.decodeLatestAggregatorValue(accountInfo);
-  if (!aggregatorAccountData)
+): { price: number; lastUpdatedSlot: number } {
+  const price = program.decodeLatestAggregatorValue(accountInfo)!.toNumber();
+  const lastUpdatedSlot = program
+    .decodeAggregator(accountInfo)
+    .previousConfirmedRoundSlot!.toNumber();
```

Comment:
hmm should this rather be latest_confirmed_round? @ckamm

Reply:
yes this matches the check in the program, updated